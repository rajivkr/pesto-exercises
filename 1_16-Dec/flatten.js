const flattenArr = (arr = []) => {
  arr = arr.reduce((a, b) => {
    return a.concat(b);
  }, []);
  return arr;
};

console.log(flattenArr([1, 2, [3, 4, [5, 6]]]));

module.exports = {
  flattenArr
};
