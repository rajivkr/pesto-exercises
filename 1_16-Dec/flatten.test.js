import { flattenArr } from './flatten';

describe('flatten array test cases', () => {
  it('should flatten the given array', () => {
    expect(flattenArr([1, 2, [3, 4, [5, 6]]])).toMatchSnapshot();
  });
});
