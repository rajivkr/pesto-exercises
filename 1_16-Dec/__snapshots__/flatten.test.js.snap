// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flatten array test cases should flatten the given array 1`] = `
Array [
  1,
  2,
  3,
  4,
  Array [
    5,
    6,
  ],
]
`;
